src/shared/psi-util.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <stdbool.h>
+
+#include "time-util.h"
+
+#define PRESSURE_CPU_FILE "/proc/pressure/cpu"
+#define PRESSURE_IO_FILE  "/proc/pressure/io"
+#define PRESSURE_MEM_FILE "/proc/pressure/memory"
+
+typedef enum PressureType {
+        PRESSURE_TYPE_SOME = 0,
+        PRESSURE_TYPE_FULL = 1,
+} PressureType;
+
+/* Averages are stored in fixed-point with 11 bit fractions */
+typedef struct ResourcePressure {
+        unsigned long avg10;
+        unsigned long avg60;
+        unsigned long avg300;
```

Comment:
i wonder if an explicit type for these wouldn't be better, i.e. 

```c
typedef unsigned long loadvg_t;
```

or so, or maybe even

```c
typedef struct {
        unsigned long value;
} loadavg_t;
```

That explains better what we are looking at and means we can't accidently access the value as integer...